android/KMEA/app/src/main/java/com/tavultesoft/kmea/KMKeyboard.java

```diff
@@ -1337,13 +1358,26 @@ public void onDismiss() {
       posY = kbPos[1] + (int) py;
     }
 
-    helpBubbleWindow.setAnimationStyle(R.style.PopupAnim);
-    if (getWindowToken() != null) {
-      helpBubbleWindow.showAtLocation(KMKeyboard.this, Gravity.TOP | Gravity.LEFT, posX, posY);
-    } else {
-      helpBubbleWindow = null;
-      ShouldShowHelpBubble = true;
-    }
+    helpBubbleWindow.showAtLocation(KMKeyboard.this, Gravity.TOP | Gravity.LEFT, posX, posY);
+  }
+
+  // This is called by KMManager to control the keyboard's size - and particularly, its height.
+  // If swapping a keyboard changes the keyboard's height (enable/disable the banner), this can
+  // cause the new position of the help bubble to be wrong.
+  @Override public void setLayoutParams(ViewGroup.LayoutParams params) {
+    super.setLayoutParams(params);
+    Log.i("HELLO", "World");
+
+    Handler handler = new Handler();
+    handler.postDelayed(new Runnable() {
+      @Override
+      public void run() {
+        if(helpBubbleWindow != null) {
```

Comment:
We don't actually need this test here because `repositionHelpBubble` does it as well.